apps/src/templates/codeDocs/ProgrammingExpressionOverview.jsx

```diff
@@ -22,21 +28,49 @@ export default function ProgrammingExpressionOverview({programmingExpression}) {
         VIDEO_HEIGHT
       );
     }
-  });
+    if (titleRef.current && programmingExpression.blockName) {
+      convertXmlToBlockly(titleRef.current);
+      const blocksDom = parseElement(
+        `<block type='${programmingExpression.blockName}' />`
+      );
+      const blockSpace = Blockly.BlockSpace.createReadOnlyBlockSpace(
+        titleRef.current,
+        blocksDom,
+        {
+          noScrolling: true,
+          inline: false
+        }
+      );
+      shrinkBlockSpaceContainer(blockSpace, true);
+    }
+  }, []);
   // Spritelab passes down its color in HSL format, whereas other labs use hex color
   const color =
     programmingExpression.color &&
     (typeof programmingExpression.color === 'string'
       ? programmingExpression.color
       : `hsl(${programmingExpression.color[0]},${programmingExpression
           .color[1] * 100}%, ${programmingExpression.color[2] * 100}%)`);
-  return (
+  const title = (
```

Comment:
this is probably personal preference, but the nested ternaries are difficult to follow -- i'd make this a function and break up the conditionals for readability